rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/RascalTextDocumentService.java

```diff
@@ -285,6 +296,56 @@ public CompletableFuture<Either<List<? extends Location>, List<? extends Locatio
             ;
     }
 
+    private ITree getCursorFromPosition(ISourceLocation file, ITree moduleTree, Position p) {
+        Position rascalCursorPos = Locations.toRascalPosition(file, p, columns);
+
+        // Find all trees containing the cursor, in ascending order of size
+        IList focusList = TreeSearch.computeFocusList(moduleTree, rascalCursorPos.getLine(), rascalCursorPos.getCharacter());
+        List<String> sortNames = focusList.stream()
+            .map(tree -> ProductionAdapter.getSortName(TreeAdapter.getProduction((ITree) tree)))
```

Comment:
nit: can be split in 2 (or 3) maps without any function closures:

```java
.map(ITree.class::cast)
.map(TreeAdapter::getProduction)
.map(ProductionAdapter::getSortName)
```

if this is better, I'm not always sure, but it does lead to less long lines?

Reply:
Agreed, replaced